src/libs/ContactUtils.ts

```diff
@@ -0,0 +1,50 @@
+import CONST from '@src/CONST';
+import type {PersonalDetails} from '@src/types/onyx';
+import type {DeviceContact, StringHolder} from './ContactImport/types';
+import * as OptionsListUtils from './OptionsListUtils';
+import {getAvatarForContact} from './RandomAvatarUtils';
+
+function sortEmailObjects(emails?: StringHolder[]): string[] {
+    if (!emails?.length) {
+        return [];
+    }
+
+    const expensifyDomain = CONST.EMAIL.EXPENSIFY_EMAIL_DOMAIN.toLowerCase();
+
+    return emails
+        .filter((email) => email?.value)
+        .map((email) => email.value)
+        .sort((a, b) => {
```

Comment:
@perunt I am not sure why we need to sort here. let's see

<img width="716" alt="Screenshot 2024-12-01 at 17 16 39" src="https://github.com/user-attachments/assets/fd84d543-58d9-48b3-9fea-d0524596d620">

we are sorting the contacts array, but in getFilteredOptions we have another logic to sort. It means that sort logic here will be overwritten

Reply:
It's for prioritizing Expensify emails if a person has several. It's meant to avoid duplicating users.

In my code, I sort emails within a contact but not the contacts themselves.
Later in the code, it filters out notification emails:
```const optionsToExclude: Option[] = [{login: CONST.EMAIL.NOTIFICATIONS}];```
I wouldn't say it contradicts itself. Or did you mean a different part of the code?